core/field_registry.js

```diff
@@ -73,8 +51,7 @@ Blockly.fieldRegistry.unregister = function(type) {
  * @package
  */
 Blockly.fieldRegistry.fromJson = function(options) {
-  var type = options['type'].toLowerCase();
-  var fieldClass = Blockly.fieldRegistry.typeMap_[type];
+  var fieldClass = Blockly.registry.getClass('field', options['type']);
```

Comment:
Why not call ``Blockly.registry.getClassFromOptions``?

Reply:
Because options here is not actually Blockly.Options. The getClassFromOptions is specifically looking for if someone has passed in the type to the plugins object.